curriculum/challenges/english/25-front-end-development/lab-blog-post-card/66eaddd04a9e533fba689001.md

```diff
@@ -0,0 +1,368 @@
+
+```js
+const card = document.querySelector("div.blog-post-card");
+assert.exists(card);
+
+const content = card.querySelector("div.post-content");
+assert.exists(content);
+
+const link = content.querySelector("a.read-more");
+assert.exists(link);
+assert.strictEqual(link.textContent, "Read More");
+```
+
+`blog-post-card` should have a `border-radius` property with a value (should not be `0` or a negative value).
+
+```js
+const card = document.querySelector('.blog-post-card');
+assert.exists(card);
+
+const borderRadius = getComputedStyle(card).borderRadius;
+assert.isAbove(parseInt(borderRadius), 0);
+```
+
+`blog-post-card` should have a white background.
+
+```js
+const card = document.querySelector('.blog-post-card');
+assert.exists(card);
+
+const backgroundColor = getComputedStyle(card).backgroundColor;
+const whiteColors = ['rgb(255, 255, 255)', 'rgba(255, 255, 255, 1)', '#ffffff', '#fff', 'white'];
+
+assert.oneOf(backgroundColor.toLowerCase(), whiteColors);
+```
+
+You should target `.blog-post-card` and set its `width` property.
+
+```js
+assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("width"));
+```
+
+You should target `.blog-post-card` and set its `text-align` property.
+
+```js
+assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("text-align"));
+```
+
+You should target `.post-content` and set its `padding` property.
+
+```js
+assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".post-content")?.getPropertyValue("padding"));
+```
+
+`.read-more` should have a hover effect.
+
+```js
+const link = document.querySelector('.read-more');
+assert.exists(link);
+
+const stylesheet = [...document.styleSheets].find(sheet => [...sheet.rules].some(rule => rule.selectorText === '.read-more:hover'));
+assert.exists(stylesheet);
+
+const hoverRule = [...stylesheet.rules].find(rule => rule.selectorText === '.read-more:hover');
+const hoverBackgroundColor = hoverRule.style.backgroundColor;
+assert.isNotEmpty(hoverBackgroundColor);
+```
+
+`.read-more` should be styled like a button. 
+
+```js
+const readMore = document.querySelector('.read-more');
+assert.exists(readMore);
+
+const readMoreDisplay = getComputedStyle(readMore).display;
+assert.strictEqual(readMoreDisplay, 'inline-block');
+
+const readMoreMargin = getComputedStyle(readMore).marginTop;
+assert.notStrictEqual(readMoreMargin, '0px');
+
+const readMorePadding = getComputedStyle(readMore).padding;
+assert.notStrictEqual(readMorePadding, '0px');
+
+const readMoreBackgroundColor = getComputedStyle(readMore).backgroundColor;
+assert.notStrictEqual(readMoreBackgroundColor, 'rgba(0, 0, 0, 0)');
+
+const readMoreTextColor = getComputedStyle(readMore).color;
+assert.notStrictEqual(readMoreTextColor, 'rgba(0, 0, 0, 0)');
+
+const readMoreBorderRadius = getComputedStyle(readMore).borderRadius;
+assert.notStrictEqual(readMoreBorderRadius, '0px');
+```
```

Comment:
It's better to write these tests separately to match the individual style in these user stories:

8. The `read-more` element should be styled like a button and have:

   - A text color other than the default.
   - Background color.
   - Margin on all sides.
   - A `display` property of `inline-block`.
   - Rounded corners.
   - Padding on all sides.